src/espresso-migration/Migration.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.8.9;
+
+import "../challenge/IChallengeManager.sol";
+import "../rollup/IRollupAdmin.sol";
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+contract Migration{ 
```

Comment:
Should this be named something like `ChallengeManagerUpgrader` or can it upgrade other contract as well?

Reply:
I agree that the contract should be renamed. I had landed on EspressoMigration as I was thinking this contract would need to be setting the wasModuleRoot. I _believe_ that the change to the wasModuleRoot actually needs to occur on the L2/orbit chain, whereas this contract would be run on the L1.